torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -134,9 +119,136 @@ def _detect_per_channel_helper(model: nn.Module):
             )
 
     if optimizations_possible:
-        further_optims_str += "To use per_channel quantization, make sure the qconfig has a per_channel weight observer."
+        further_optims_str += (
+            "To use per_channel quantization, make sure the qconfig has a per_channel weight observer."
+        )
     else:
         further_optims_str += "No further per_channel optimizations possible."
 
     # return the string and the dictionary form of same information
     return (further_optims_str, per_channel_info)
+
+
+# names for the pre and post observers that are inserted
+DEFAULT_PRE_OBSERVER_NAME = "model_report_pre_observer"
+DEFAULT_POST_OBSERVER_NAME = "model_report_post_observer"
+
+# naming conventions for stationary vs non-stationary data
+DEFAULT_STATIONARY = "stationary"
+DEFAULT_NON_STATIONARY = "non-stationary"
+
+
+def _detect_dynamic_vs_static(model: GraphModule, tolerance=0.5) -> Tuple[str, Dict[str, Any]]:
```

Comment:
Is there any guidance on how to pick this value? I guess 0.5 is a reasonable one but I wonder if it's hard for the user to reason about what a good threshold is

Reply:
There really isn't for now, which is why we just decided to make it a parameter that the user could choose.